app/src/main/java/org/simple/clinic/summary/bloodpressures/BloodPressureSummaryViewEffectHandler.kt

```diff
@@ -9,13 +10,15 @@ import org.simple.clinic.bp.BloodPressureRepository
 import org.simple.clinic.facility.FacilityRepository
 import org.simple.clinic.user.UserSession
 import org.simple.clinic.util.scheduler.SchedulersProvider
+import javax.inject.Named
 
 class BloodPressureSummaryViewEffectHandler @AssistedInject constructor(
     private val userSession: UserSession,
     private val facilityRepository: FacilityRepository,
     private val bloodPressureRepository: BloodPressureRepository,
     private val schedulersProvider: SchedulersProvider,
-    @Assisted private val uiActions: BloodPressureSummaryViewUiActions
+    @Assisted private val uiActions: BloodPressureSummaryViewUiActions,
+    @Named("is_facility_switched") private val isFacilitySwitchedPreference: Preference<Boolean>
```

Comment:
While it doesn't make sense to use `Boolean` values for states, I think this is a valid use case for it. Storing `Optional<Instant>` will add unnecessary complexity. I don't see how it improves the code. I will still have to filter everywhere.

Reply:
It's not about changing the ease of implementation, but about changing the meaning of the type.

Boolean types are easy to misuse since the name of the type and the way it is used can get out of sync and cause errors.